src/sentry/api/serializers/models/release.py

```diff
@@ -311,16 +319,41 @@ def get_attrs(self, item_list, user, **kwargs):
         if with_health_data and no_snuba:
             raise TypeError("health data requires snuba")
 
+        if self.with_adoption_stages:
+            release_project_envs = (
+                ReleaseProjectEnvironment.objects.filter(release__in=item_list)
+                .select_related("release", "project")
+                .order_by("-first_seen")
+            )
+            if environments is not None:
+                release_project_envs = release_project_envs.filter(
+                    environment__name__in=environments
+                )
+            if project is not None:
+                release_project_envs = release_project_envs.filter(project=project)
+
+            adoption_stages = self._get_release_adoption_stages(release_project_envs)
+
         if environments is None:
             first_seen, last_seen, issue_counts_by_release = self.__get_release_data_no_environment(
                 project, item_list
             )
         else:
+            if not self.with_adoption_stages:
```

Comment:
It might be better to set `release_project_envs = None` earlier, than check if it's `None` here, rather than relying on the flag.